drivers/ipcc/ipcc_write.c

```diff
@@ -239,15 +239,20 @@ ssize_t ipcc_write(FAR struct file *filep, FAR const char *buffer,
        */
 
       nxsem_post(&priv->exclsem);
+
       if ((ret = nxsem_wait(&priv->txsem)))
         {
+          leave_critical_section(flags);
+
           /* We were interrupted by signal, return error or number
            * of bytes written
            */
 
           return nwritten ? nwritten : -EINTR;
         }
 
+      leave_critical_section(flags);
```

Comment:
Maybe just place `flags = enter_critical_section();` before `for (nwritten = 0; ; )` instead of adding `leave_critical_section(flags);` here?

Reply:
Done